script/gen-theme.js

```diff
@@ -0,0 +1,193 @@
+  ],
+  "button-active-shadow": [
+    "0 0.33px 0 1px #ffffff26, 0 4px 6px -3px #00000026, 0 0.4px 0 0.8px #00000073",
+    "0 -0.33px 0 1px #ffffff44, 0 4px 6px -3px #00000026, 0 0.4px 0 0.8px #00000073",
+    "0 -0.33px 0 1px #ffffff44, 0 4px 6px -3px #00000026, 0 0.4px 0 0.8px #00000073",
+  ],
+  "button-inset-shadow": [
+    "0 0.33px 0 1px #ffffff26 inset, 0 4px 8px -3px #00000026 inset, 0 0.4px 0 0.8px #00000040 inset",
+    "0 -0.33px 0 1px #ffffff44 inset, 0 4px 8px -3px #00000026 inset, 0 0.4px 0 0.8px #00000040 inset",
+    "0 -0.33px 0 1px #ffffff44 inset, 0 4px 8px -3px #00000026 inset, 0 0.4px 0 0.8px #00000040 inset",
+  ],
+
+  // Shadows
+  "heavy-3d-box-shadow": [
+    "0 0.33px 0 1px #ffffff26, 0 8px 12px -6px #0000004d, 0 0.5px 0 1px #00000033",
+    "0 -0.33px 0 1px #ffffff26, 0 8px 12px -6px #0000004d, 0 0.5px 0 1px #00000033",
+    "0 -0.33px 0 1px #ffffff26, 0 8px 12px -6px #0000004d, 0 0.5px 0 1px #00000033",
+  ],
+  "text-input-shadow": [
+    "0 0.33px 0 1px #ffffff26, 0 6px 9px -4px #0000001a inset, 0 0.4px 0 0.8px #0000001a inset",
+    "0 -0.33px 0 1px #ffffff26, 0 6px 9px -4px #0000001a inset, 0 0.4px 0 0.8px #0000001a inset",
+    "0 -0.33px 0 1px #ffffff26, 0 6px 9px -4px #0000001a inset, 0 0.4px 0 0.8px #0000001a inset",
+  ],
+  "dropdown-shadow": [
+    "0 0.33px 0 1px #ffffff26, 0 12px 12px -4px #00000033, 0 0.4px 0 0.8px #00000059",
+    "0 -0.33px 0 1px #ffffff55, 0 12px 12px -4px #00000033, 0 0.4px 0 0.8px #00000059",
+    "0 -0.33px 0 1px #ffffff55, 0 12px 12px -4px #00000033, 0 0.4px 0 0.8px #00000059",
+  ],
+};
+
+const childVariables = {
+  "log-color": [
+    "var(--light-log-color)",
+    "var(--dark-log-color)",
+    "var(--dark-log-color)",
+  ],
+  "log-bg-color": [
+    "var(--light-log-bg-color)",
+    "var(--dark-log-bg-color)",
+    "var(--dark-log-bg-color)",
+  ],
+  "log-hover-color": [
+    "var(--light-log-hover-color)",
+    "var(--dark-log-hover-color)",
+    "var(--dark-log-hover-color)",
+  ],
+  "log-bg-hover-color": [
+    "var(--light-log-bg-hover-color)",
+    "var(--dark-log-bg-hover-color)",
+    "var(--dark-log-bg-hover-color)",
+  ],
+};
+
+// Format a list of root level theme variables.
+const themeDefinition = (theme) =>
+  Object.entries(themeVariables)
+    .map((entry) => {
+      return `  --${entry[0]}: ${entry[1][theme]};`;
+    })
+    .join("\n");
+
+// Format a list of child level theme variables.
+const childDefinition = (theme) =>
+  Object.entries(childVariables)
+    .map((entry) => {
+      return `  --${entry[0]}: ${entry[1][theme]};`;
+    })
+    .join("\n");
+
+// Format a block of theme variables for root and child selectors.
+const themeBlock = (selector, theme) => `body.${selector} {
+${themeDefinition(theme)}
+}
+body.${selector} * {
+${childDefinition(theme)}
+}`;
+
+// Format the entire theme file using the above helpers.
+const out = `/* Generated file. DO NOT EDIT. */
```

Comment:
not that it matters much for a small code generator like this, but in the future, you should probably perform code generation imperatively by doing `out += ...` instead of using string interpolation to build up a large string. Otherwise, you wind up copying the strings over and over and over again as you iteratively rebuild immutable strings up the tree of interpolations.